paasta_tools/kubernetes_tools.py

```diff
@@ -3432,3 +3434,44 @@ def update_crds(
             success = False
 
     return success
+
+
+def get_kubernetes_secret_name(service_name, secret_name, namespace="paasta"):
+    service = sanitise_kubernetes_name(service_name)
+    sanitised_secret = sanitise_kubernetes_name(secret_name)
+    name = f"{namespace}-secret-{service}-{sanitised_secret}"
+    return name
+
+
+def get_kubernetes_secret(secret_name, service_name, cluster):
+    k8s_secret_name = get_kubernetes_secret_name(service_name, secret_name)
+    template_name = f"{{.data.{secret_name}}}"
+
+    env = os.environ.copy()
+    env["KUBECONFIG"] = DEFAULT_KUBECTL_CONFIG_PATH
+
+    proc = subprocess.run(
+        [
+            "kubectl",
+            "get",
+            "secrets",
+            f"--cluster={cluster}",
+            f"-n=paasta",
+            f"{k8s_secret_name}",
+            f"--template={{{template_name}}}",
+        ],
+        stderr=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        env=env,
+    )
+
+    # Error running kubectl
+    if proc.returncode != 0:
+        process_errors = proc.stderr.decode("utf-8").strip()
+        if process_errors:
+            print(process_errors)
+        return "Could not get secret.\n"
```

Comment:
Sorry for not catching it in the previous review, this should probably just throw an exception.